### DISCUSSION TITLE
If the build process uses the licx file to check what licenses shoud/can be embedded into the output.. how does an empty file help?

### DISCUSSION THREAD

**Discussion by dave-q** [augustoproiete/EmptyLicensesLicx#2] [Q&A] [answered]
This is a question that I'm hoping someone can help with.
Here is a breakdown of my understanding

* Licx file is modified by VS everytime I utilize a control, it adds an entry to say this control is licensed
* At build time it takes this list of licenses controls in the licx file and checks the license on the building machine and if its there, embeds the licenses into a resource in the outputted build
* If the license is not found you get an error
* If there are no entries... you get no embedded licenses (I assume)

So how does having an empty licences.licx file solve things... won't it just produce a build that doesn't have any embedded licenses?

if those embedded licenses aren't important.. whats the point of all of this?

**Comment by augustoproiete** [comment, marked as answer]
Hi @dave-q, that's a good question.

It all depends on how the licensing of the components you're using work. Most .NET control vendors such as DevExpress, Telerik, Infragistics, ComponentOne, etc. **do not require a license for _distributing_ your application** (and there's no license check when your application runs in `Release` mode), so you don't need to embed any licenses with your apps during build.

These vendors require you to have a valid license for **development purposes only**, and the license check is enforced during development time only, usually when you're designing your UI (e.g. using the Visual Studio designer), and/or when you're running your app in `Debug` mode.

That's why the empty `Licenses.licx` works for the purposes of building your app, and it also saves you from having to install the vendor control suite on your build server.

Does that make sense?

That said, if you are using a third-party control that requires a **runtime** license (i.e. the license is required to run your app on any machine) then the empty `Licenses.licx` doesn't make sense for you, and you do need to embed the licenses with your app during the build... This is not something I've seen in recent years, but I'm sure it still exists.

**Comment by dave-q** [comment]
Hi,

Thanks for the quick reply, its really appreciated. That has really cleared things up thank you.

We use DevExpress v19 (no runtime licensing), but we also use Syncfusion v6.4 (has run time licensing ) :( , so it looks like we are back to checking in the license.licx file.

If we ever upgrade I'll be sure to come and use this. Looks like a very useful tool
